### PR TITLE
Method BANKSY

### PR DESCRIPTION
**Pull request by shdam**
Implemented Banksy (issue #8)

I don't see a way to not use the method's own way to compute neighbors, so those are used.
It also needs both PCA and expression values, so I use its own method for PCA.

### CODE REVIEW

File: method/BANKSY/banksy_env.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# Create the BANKSY conda environment named banksy_env
+conda env create -f banksy.yml
+
+# Activate the environment
+conda activate banksy_env
+
+# Install the required R packages
+Rscript -e "remotes::install_github('prabhakarlab/Banksy', dependencies = TRUE, ref = 'b1a2c8bb2af06346f303637b9bba18faa1a1fe32')"
```
Comment (niklasmueboe) [inline method/BANKSY/banksy_env.sh:10]: Can we use a release tag here instead of a commit hash?

File: method/BANKSY/banksy.r [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+## Your code goes here
+lambda <- config$lambda
+k_geom <- config$k_geom
+npcs <- config$npcs
+resolution <- config$resolution
+method <- config$method
+assay_name <- "logcounts"
+set.seed(seed)
+spe <- Banksy::computeBanksy(spe, assay_name = assay_name, k_geom = k_geom)
+set.seed(seed)
```
Comment (niklasmueboe) [inline method/BANKSY/banksy.r:174]: duplicate code can be removed

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

**Comment by niklasmueboe** [comment]
@svedziok can you review the conda env generation?

File: method/BANKSY/banksy_env.sh (hunk 1 above)
Comment (shdam) [reply method/BANKSY/banksy_env.sh]: Done

File: method/BANKSY/banksy.r (hunk 2 above)
Comment (shdam) [reply method/BANKSY/banksy.r]: done

**Comment by svedziok** [comment]
> @svedziok can you review the conda env generation?
 looks fine; we will then add an additional rule for installing the missing package for this method

**Review by niklasmueboe — APPROVED** [review]

**Pull request merged**